src/modules/LanItems.ycp

```diff
@@ -1831,28 +1832,49 @@ global map FindMatchingDevice(map interface) {
     return tosel;
 }
 
-global void DeleteItem(){
- y2milestone("deleting ... %1", Items[current]:$[]);
- string ifcfg = Items[current, "ifcfg"]:"";
- string hwcfg = Items[current, "hwcfg"]:"";
+global void DeleteItem()
+{
+    y2milestone("deleting ... %1", Items[current]:$[]);
 
- if (IsNotEmpty(ifcfg)) {
-	 NetworkInterfaces::Delete(ifcfg);
-	 NetworkInterfaces::Commit();
-	 Items[current, "ifcfg"] = "";
-	}
- if (! (size(Items[current, "hwinfo"]:$[])>0) ){
-  map<integer, any> tmp_items = $[];
-  foreach(integer key, any value, Items, {
-   if (key==current) continue;
-    else{
-	if (key<current) tmp_items[key]=Items[key]:$[];
-		else tmp_items[key-1]=Items[key]:$[];
-	}
-  });
-  Items = tmp_items;
- }
- SetModified();
+    string ifcfg = Items[current, "ifcfg"]:"";
+    string hwcfg = Items[current, "hwcfg"]:"";
+
+    if (IsNotEmpty(ifcfg))
+    {
+        NetworkInterfaces::Delete(ifcfg);
+
+        // this depends on above NetworkInterfaces::Delete
```

Comment:
> NetworkInterfaces::Delete sets NetworkInterfaces::current internally, which is used in GetValue

While both statements are true, they don't imply that `GetValue` needs `Delete`. [`GetValue`](https://github.com/yast/yast-yast2/blob/7b9adc08fe956778c9f112a0c095a089c6478194/library/network/src/NetworkInterfaces.ycp#L1567) has ifcfg as an argument, and does call `Select` on it.

> Moreover, you want to delete ifcfg even when no IP is present (e.g. dhcp config) so it doesn't hurt.

Sure, I mean like this:
```cpp
+        // no comment needed here
+        string ip = NetworkInterfaces::GetValue(ifcfg, "IPADDR");
+        if(IsNotEmpty(ip))
+        {
+            y2milestone("Deleting hostnames assigned to %1 from /etc/hosts", ip);
+            Host::SetNames(ip, []);
+        }
+        NetworkInterfaces::Delete(ifcfg);
```